SonarQube.MSBuild.Tasks/Resources.resx

```diff
@@ -193,4 +193,19 @@ Error: {1}</value>
   <data name="AnalyzerSettings_NotSpecifiedInConfig" xml:space="preserve">
     <value>Analyzer settings have not been specified in the analysis config file</value>
   </data>
+  <data name="BuildWrapper_AlreadyAttached" xml:space="preserve">
+    <value>BuildWrapper: the build wrapper is already attached to the current process ({0})</value>
+  </data>
+  <data name="BuildWrapper_AttachedSuccessfully" xml:space="preserve">
+    <value>BuildWrapper: the build wrapper attached successfully</value>
+  </data>
+  <data name="BuildWrapper_FailedToAttach" xml:space="preserve">
+    <value>BuildWrapper: failed to attach to the current process</value>
+  </data>
+  <data name="BuildWrapper_RequiredFileMissing" xml:space="preserve">
```

Comment:
Not really. If we hit this then it means something has gone wrong in our code somewhere else. This is just an attempt to give a slightly more useful error message to help diagnose the issue.
The pre-processor is responsible for working out if the C++ plugin is installed and has an instance of the build wrapper (i.e. is a new enough version of the C++ plugin).
The targets are responsible for working out if the build wrapper is available on disk and only calling the task if it is. 
The task then checks that the "file contract" with the C++ plugin has been met i.e. the expected files exist. If not, it probably means that the C++ plugin is invalid i.e. it's broken the file contract.